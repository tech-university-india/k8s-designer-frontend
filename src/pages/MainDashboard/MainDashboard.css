.main-dashboard {
  display: flex;
  flex-direction: column;
  /* width: fit-content; */
}

.dashboard-service {
  display: flex;
  height: 100vh;
  width: 100vw;
}

.simple-floatingedges {
  flex-direction: column;
  display: flex;
  flex-grow: 1;
  height: 100%;
}

.simple-floatingedges .react-flow__handle {
  width: 8px;
  height: 8px;
  background-color: #bbb;
}

.simple-floatingedges .react-flow__handle-top {
  top: -15px;
}

.simple-floatingedges .react-flow__handle-bottom {
  bottom: -15px;
}

.simple-floatingedges .react-flow__handle-left {
  left: -15px;
}

.simple-floatingedges .react-flow__handle-right {
  right: -15px;
}

.simple-floatingedges .react-flow__node-custom {
  background: #fff;
  border: 1px solid #1a192b;
  border-radius: 3px;
  color: #222;
  font-size: 12px;
  padding: 10px;
  text-align: center;
}

.react-flow__node-custom {
  border: 2px solid black;
  border-radius: 2px;
  padding: 10px;
}

.config-body{
  border: 1px solid rgb(82, 81, 81);
  height: 100vh;
  width: 30vw;
  background-color: white;
  margin: 0;
  font-family: 'Open Sans', sans-serif;
  font-weight: 800;
  overflow: auto;
}
.config-element-border{
  height: 5vh;
  font-size: 20px;
  border-bottom : 2px solid rgb(82, 81, 81);
  text-align: center;
  padding-top: 2vh;
  
}

@media screen and (max-width:768px) {
  .config-element-border{
      font-size: 15px;
      height: 5vh;
      padding-top: 1.5vh;

  }
  
}