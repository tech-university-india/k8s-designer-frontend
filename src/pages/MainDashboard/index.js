import React, { useState, useRef, useCallback } from 'react';
import { v4 as uuidv4 } from 'uuid';
import './MainDashboard.css';
import { Header, MicroServices } from '../../components';
import ReactFlow, {
  addEdge,
  useNodesState,
  useEdgesState,
  Background,
} from 'reactflow';

// import 'reactflow/dist/style.css';
import { CustomNode } from '../../utils';

const nodeTypes = {
  microservice: CustomNode,
};

const MainDashboard = () => {
  const reactFlowWrapper = useRef(null);
  const [nodes, setNodes, onNodesChange] = useNodesState([]);
  const [edges, setEdges, onEdgesChange] = useEdgesState([]);
  const [reactFlowInstance, setReactFlowInstance] = useState(null);

  const onConnect = useCallback((params) => setEdges((eds) => addEdge({ ...params, type: 'microserice' }, eds)), []);

  const onDragOver = useCallback((event) => {
    event.preventDefault();
    event.dataTransfer.dropEffect = 'move';
  }, []);

  const onDrop = useCallback(
    (event) => {
      event.preventDefault();

      const reactFlowBounds = reactFlowWrapper.current.getBoundingClientRect();
      const type = event.dataTransfer.getData('application/reactflow');
      const name = event.dataTransfer.getData('nameOfNode');

      console.log(type);
      console.log(name);

      if (typeof type === 'undefined' || !type) {
        return;
      }

      const position = reactFlowInstance.project({
        x: event.clientX - reactFlowBounds.left,
        y: event.clientY - reactFlowBounds.top,
      });
      const newNode = {
        id: uuidv4(),
        type,
        position,
        data: { label: name.toUpperCase() },
      };
      setNodes((nds) => {return  [...nds, newNode];});
    },
    [reactFlowInstance]
  );
  return (
    <div className='maindashboard'>
      <Header />
      <div className='dashboard_service' ref={reactFlowWrapper}>
        <MicroServices />
        <ReactFlow
          nodes={nodes}
          edges={edges}
          onNodesChange={onNodesChange}
          onEdgesChange={onEdgesChange}
          onConnect={onConnect}
          nodeTypes={nodeTypes}
          onInit={setReactFlowInstance}
          onDrop={onDrop}
          onDragOver={onDragOver}
          fitView
          fitViewOptions={ { padding: 4 } }
          connectionMode="loose"
        >
          <Background color="#aaa" gap={16} />
        </ReactFlow>
      </div>
    </div>
  );
};

export default MainDashboard;