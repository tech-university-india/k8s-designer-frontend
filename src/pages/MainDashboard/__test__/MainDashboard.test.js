import React from 'react';
import { render } from '@testing-library/react';
import MainDashboard from './../index';

import * as ResizeObserverModule from 'resize-observer-polyfill';

(global).ResizeObserver = ResizeObserverModule.default;

jest.mock('reactflow/dist/style.css', () => {});
describe('MainDashboard', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<MainDashboard />);
    expect(asFragment()).toMatchSnapshot();
  });
});