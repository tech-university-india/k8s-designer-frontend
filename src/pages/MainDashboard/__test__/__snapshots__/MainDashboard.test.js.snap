// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainDashboard should render correctly 1`] = `
<DocumentFragment>
  <div
    class="main-dashboard"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        K8D
      </div>
      <div
        class="user"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Logout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="dashboard-service"
    >
      <div
        class="microservices"
      >
        <div
          class="microservices-header"
        >
          MICROSERVICES
        </div>
        <div
          class="services"
        >
          <div
            class="service"
            draggable="true"
          >
            <p>
              FRONTEND
            </p>
          </div>
          <div
            class="service"
            draggable="true"
          >
            <p>
              BACKEND
            </p>
          </div>
          <div
            class="service"
            draggable="true"
          >
            <p>
              DATABASE
            </p>
          </div>
        </div>
      </div>
      <div
        class="react-flow"
        data-testid="rf__wrapper"
        style="width: 100%; height: 100%; overflow: hidden; position: relative; z-index: 0;"
      >
        <div
          class="react-flow__renderer"
          style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
        >
          <div
            class="react-flow__pane"
            style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
          >
            <div
              class="react-flow__viewport react-flow__container"
              style="transform: translate(0px,0px) scale(1);"
            >
              <svg
                class="react-flow__edges react-flow__container"
                height="500"
                style="z-index: 0;"
                width="500"
              >
                <defs />
                <g />
              </svg>
              <div
                class="react-flow__edgelabel-renderer"
              />
              <div
                class="react-flow__nodes"
                style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
              />
            </div>
          </div>
        </div>
        <svg
          class="react-flow__background"
          style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
        >
          <pattern
            height="16"
            id="pattern-1"
            patternTransform="translate(-0.5,-0.5)"
            patternUnits="userSpaceOnUse"
            width="16"
            x="0"
            y="0"
          >
            <circle
              cx="0.5"
              cy="0.5"
              fill="#aaa"
              r="0.5"
            />
          </pattern>
          <rect
            fill="url(#pattern-1)"
            height="100%"
            width="100%"
            x="0"
            y="0"
          />
        </svg>
        <div
          class="react-flow__panel react-flow__attribution bottom right"
          data-message="Please only hide this attribution when you are subscribed to React Flow Pro: https://pro.reactflow.dev"
          style="pointer-events: all;"
        >
          <a
            aria-label="React Flow attribution"
            href="https://reactflow.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            React Flow
          </a>
        </div>
        <div
          id="react-flow__node-desc-1"
          style="display: none;"
        >
          Press enter or space to select a node.You can then use the arrow keys to move the node around. Press delete to remove it and escape to cancel. 
        </div>
        <div
          id="react-flow__edge-desc-1"
          style="display: none;"
        >
          Press enter or space to select an edge. You can then press delete to remove it or escape to cancel.
        </div>
        <div
          aria-atomic="true"
          aria-live="assertive"
          id="react-flow__aria-live-1"
          style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(100%);"
        />
      </div>
      <div
        class="config-body"
      >
        <div
          class="config-element-border"
        >
          CONFIGURATION
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
